src/node.rs

```diff
@@ -1995,6 +2014,17 @@ impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> EthNamespaceT for
             .boxed();
         };
 
+        match self.validate_tx(&l2_tx) {
```

Comment:
We should move this a layer down to `run_l2_tx(..)` as not all transactions are coming from `send_raw_transaction` endpoint.

Right before we run `self.run_l2_tx_inner(...)`, we should add:
```rust
        match self.validate_tx(&l2_tx) {
            Ok(_) => (),
            Err(e) => {
                return Err(e);
            }
        };
```

Reply:
Had it initially there, but moved out as currently `run_l2_tx` errors are processed as `Execution` errors and I think this one should not be an `Execution` error, also it seems like validating just `send_raw_transaction` is enough as all transactions apart from those from fork should call  `send_raw_transaction`. For those from fork I don't think we have to validate them as they were already executed and should always be valid.

Reply:
It's the case for unit tests. Feel free to add unit tests that uses this validation for the transaction and see what you come up with